android-lib/src/main/java/com/kinvey/android/store/DataStore.kt

```diff
@@ -418,6 +416,39 @@ open class DataStore<T : GenericJson> : BaseDataStore<T> {
         createBatch(entities, callback)
     }
 
+    /**
+     * Asynchronous request to create an entity to a collection.
+     *
+     *
+     * Constructs an asynchronous request to create an entity <T> to a collection.
+     * Creates the entity if it doesn't exist, return error for entity if it does exist.
+     * If an "_id" property is not present, the Kinvey backend will generate one.
+    </T> *
+     *
+     *
+     * Sample Usage:
+     * <pre>
+     * `DataStore<EventEntity> myAppData = DataStore.collection("myCollection", EventEntity.class, StoreType.SYNC, myClient);
+     * myAppData.create(entity, new KinveyClientCallback<EventEntity> {
+     * public void onFailure(Throwable t) { ... }
+     * public void onSuccess(EventEntity entity) { ... }
+     * });
+    ` *
+    </pre> *
+     *
+     *
+     * @param entity The entity to create
+     * @param callback KinveyClientCallback<T>
+    </T> */
+    fun create(entity: T, callback: KinveyClientCallback<T>) {
+        Preconditions.checkNotNull(client, "client must not be null")
+        Preconditions.checkArgument(client?.isInitialize ?: false, "client must be initialized.")
+        Preconditions.checkNotNull(entity, "Entity cannot be null.")
+        Preconditions.checkState(kinveyApiVersion >= KINVEY_API_VERSION_5, "Kinvey api version cannot be less than 5.")
```

Comment:
Actually for single-entity insert all api versions would work. This check should only be done when calling `create` with an array of entities.

Reply:
Fixed